application/src/main/kotlin/net/corda/v5/application/uniqueness/client/UniquenessCheckerClientService.kt

```diff
@@ -0,0 +1,55 @@
+package net.corda.v5.application.uniqueness.client
+
+import net.corda.data.uniqueness.UniquenessCheckResponse
+import net.corda.v5.base.annotations.DoNotImplement
+import java.time.Instant
+import java.util.concurrent.Future
+
+/**
+ * This service is used to queue a request for uniqueness checking.
+ * The request will be processed asynchronously and a [Future] will
+ * be returned that can be used to check the outcome.
+ *
+ * This service can be injected to either flows or other services.
+ *
+ * The actual request processing depends on which uniqueness service
+ * is invoked from the client.
+ *
+ * The API contains only one function that is used to queue up a request.
+ */
+@DoNotImplement
+interface UniquenessCheckerClientService {
+
+    /**
+     * This function is used to queue up a request for the client service,
+     * which will process it in an async manner, and return a [Future].
+     *
+     * @param txId The ID of the transaction that needs processing
+     *
+     * @param inputs A list of the input state refs that belongs
```

Comment:
Line lengths seem very conservative in this file; using 100 characters would probably reduce the number of lines in the file.